bip-0000.mediawiki

```diff
@@ -0,0 +1,206 @@
+
+* Let ''b<sub>spend</sub> = m/BIP’/0’/k''
+* Let ''b<sub>scan</sub> = m/BIP’/1’/k''
+
+Each value of ''k'' represents a pair of private keys needed for creating a silent payment address. In this way, multiple silent payment addresses can be derived from a single master key. Wallet software MUST use hardened derivation to ensure the master key is not exposed in the event the scan private key is compromised.
+
+==== Scanning ====
+
+For each transaction the receiving wallet suspects might be a silent payment to themselves, it must<ref name="data_needed_for_scanning>'''What data does a wallet need for scanning''' A wallet needs a 32 byte public key which represents the sum of the input public keys of a transaction, the truncated 8 byte hash of the transaction input outpoints, and some method of determining whether or not a scriptPubKey exists in a block. For full nodes, all of this data is already available. For light clients, they would need to request the tweak data from a server with access to the necessary information for tweaking and determine existence in a block using something like BIP158. BIP158 is recommended over other methods as this is the most trust minimized, privacy preserving method for light clients today.</ref>:
+
+* Generate the ''outpoint_hash'', using the method described above
+* Let ''A = A<sub>0</sub> + A<sub>1</sub> + … A<sub>i</sub>'', where each ''A<sub>i</sub>'' is the public key corresponding to the ''ith'' input of the transaction
+* Compute the tweaks:
+** Let ''M = the number of labels the Recipient wallet has used''
+** Let ''t<sub>0</sub> = SHA256(SHA256(outpoint_hash·b<sub>scan</sub>·A || 0))''
+** Compute ''P<sub>00</sub> = t<sub>0</sub>·G + B<sub>spend</sub>''
+** If ''M'' is greater than 0:
+*** Let ''m = 1'':
+*** While ''m'' &lt;= ''M'':
+**** Compute ''P<sub>m0</sub> = P<sub>00</sub> + B<sub>spend</sub> + m·G''
+**** repeat with ''m++''
+** For each public key ''P<sub>m0</sub>'' in {''P<sub>00</sub> … P<sub>M0</sub>''}:
+*** If ''P<sub>m0</sub>'' is detected in the transaction outputs, starting with ''n = 1'':
+**** Compute ''t<sub>n</sub> = SHA256(SHA256(outpoint_hash·b<sub>scan</sub>·A || n))''
+**** Check if ''P<sub>mn</sub> = (t<sub>n</sub> + m)·G + B<sub>spend</sub>'' is present in the remaining outputs
+**** If not, terminate
+**** If ''P<sub>mn</sub>'' is found, repeat with ''n++''
+
+For each output with public key ''P<sub>mn</sub>'', store the output, label ''m'', and tweak data ''t<sub>n</sub>'' in the wallet. ''P<sub>mn</sub>'' is spendable with the private key ''x = t<sub>n</sub> + m + b<sub>spend</sub> mod p''
+
+==== Backup and Recovery ====
+
+Since each silent payment output address is derived independently, we cannot rely on BIP32 derivation when recovering from a backup. Because of this, descriptor wallets are recommended along with regular backups. When recovering from a backup, the wallet will need to scan since the last backup to detect new payments.
```

Comment:
```suggestion
Since each silent payment output address is derived independently, descriptor wallets are recommended along with regular backups. When recovering from a backup, the wallet will need to scan since the last backup to detect new payments.
```
Just a suggestion, but I feel the BIP32 info probably is only more likely to confuse people than be informative. And perhaps in general this section can use a bit more elaboration.